### PR TITLE
run the benchmark from a tupelo command

### PR DESCRIPTION
**Pull request by tobowers**
this brings back the benchmark command. I tested using the docker-compose-benchmark.yml like so: `docker-compose -f docker-compose.yml -f docker-compose-benchmark.yml up`

### CODE REVIEW

**Comment by cap10morgan** [comment]
Oh, haha. I started on this too b/c the card was unclaimed in Trello.

File: cmd/benchmark.go [hunk 1]
```diff
@@ -1,267 +1,81 @@
...
+		fmt.Println(string(out))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(benchmarkCmd)
+	benchmarkCmd.Flags().IntVarP(&benchmarkConcurrency, "concurrency", "c", 1, "how many transactions to execute at once")
+	benchmarkCmd.Flags().IntVarP(&benchmarkDuration, "duration", "d", 10, "how many seconds to run benchmark for")
+	benchmarkCmd.Flags().IntVarP(&benchmarkTimeout, "timeout", "t", 0, "seconds per transaction to wait before timing out")
+	benchmarkCmd.Flags().IntVar(&benchmarkStartDelay, "delay", 0, "how many seconds to wait before kicking off the benchmark; useful if network needs to stablize first")
```
Comment (cap10morgan) [inline cmd/benchmark.go]: "stabilize" is misspelled in the help text (was before too; likely my fault originally)

File: cmd/benchmark.go [hunk 2]
```diff
@@ -1,267 +1,81 @@
...
+		out, _ := json.MarshalIndent(results, "", "  ")
+		fmt.Println(string(out))
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(benchmarkCmd)
+	benchmarkCmd.Flags().IntVarP(&benchmarkConcurrency, "concurrency", "c", 1, "how many transactions to execute at once")
+	benchmarkCmd.Flags().IntVarP(&benchmarkDuration, "duration", "d", 10, "how many seconds to run benchmark for")
+	benchmarkCmd.Flags().IntVarP(&benchmarkTimeout, "timeout", "t", 0, "seconds per transaction to wait before timing out")
```
Comment (cap10morgan) [inline cmd/benchmark.go]: Doesn't look like this gets used anywhere. Probably fine to leave it out for now until / unless we need it.

File: cmd/benchmark.go [hunk 3]
```diff
@@ -1,267 +1,81 @@
...
+var benchmarkConcurrency int
+var benchmarkTimeout int
+var benchmarkDuration int
+var benchmarkStartDelay int
+
+// benchmark represents the shell command
+var benchmarkCmd = &cobra.Command{
+	Use:    "benchmark",
+	Short:  "runs a set of operations against a network at specified concurrency",
+	Hidden: true,
```
Comment (cap10morgan) [inline cmd/benchmark.go]: What does this do? Do we want it?

File: cmd/benchmark.go [hunk 4]
```diff
@@ -1,267 +1,80 @@
...
+
+var benchmarkConcurrency int
+var benchmarkTimeout int
+var benchmarkDuration int
+var benchmarkStartDelay int
+
+// benchmark represents the shell command
+var benchmarkCmd = &cobra.Command{
+	Use:    "benchmark",
+	Short:  "runs a set of operations against a network at specified concurrency",
```
Comment (cap10morgan) [inline cmd/benchmark.go]: I added "and duration" in my start on this. Not a big deal either way.

File: cmd/benchmark.go [hunk 5]
```diff
@@ -1,267 +1,80 @@
...
+	"github.com/quorumcontrol/tupelo-go-sdk/p2p"
+	"github.com/quorumcontrol/tupelo/gossip/benchmark"
+	"github.com/quorumcontrol/tupelo/nodebuilder"
+	"github.com/spf13/cobra"
+)
+
+var benchmarkConcurrency int
+var benchmarkTimeout int
+var benchmarkDuration int
+var benchmarkStartDelay int
```
Comment (cap10morgan) [inline cmd/benchmark.go]: I removed the redundant `benchmark` prefix from all of these in my start on this. Take or leave it. :)

Comment (tobowers) [reply cmd/benchmark.go]: well - the vars aren't local to this file, they are local to the "cmd" package and so if any other packages had something like "timeout" then that could be confusing.

File: cmd/benchmark.go (hunk 3 above)
Comment (tobowers) [reply cmd/benchmark.go]: keeps it out of help (this was in the original benchmark)... not sure if we want it in or out.

File: cmd/benchmark.go (hunk 5 above)
Comment (cap10morgan) [reply cmd/benchmark.go]: Ah, yeah, good point. 👍 to leaving the prefix then.

File: cmd/benchmark.go (hunk 3 above)
Comment (cap10morgan) [reply cmd/benchmark.go]: OK, makes sense. Yeah, I don't have a strong opinion either way.

**Comment by tobowers** [comment]
> Mostly fine, just a few minor things to consider and one unused var that we should probably do something with or remove (`benchmarkTimeout`).

I think I hit all the areas... left Hidden in there now as a cargo cult mostly

**Pull request merged**